Pack.cs

```diff
@@ -4,30 +4,115 @@
 using System.Text;
 using System.Threading.Tasks;
 
-namespace CMP1903M_A01_2223
+namespace CMP1903MA012223
 {
     class Pack
     {
-        List<Card> pack;
+
+        public static List<Card> pack = new List<Card>();
+        public static List<Card> split_deck = new List<Card>();
+        public static List<Card> drawncards = new List<Card>();
+        public static List<Card> dealt_cards = new List<Card>();
 
         public Pack()
         {
-            //Initialise the card pack here
-        }
+            //Initialise the card pack here
 
-        public static bool shuffleCardPack(int typeOfShuffle)
+            int i = 1;
+            int j = 1;
+
+            while (j != 5)
+            {
+                pack.Add(new Card(i, j));
+                i++;
+                if (i == 13)
+                {
+                    i = 1;
+                    j++;
+                }
+            }
+        }
+        
+        public static bool ShuffleCardPack(int typeOfShuffle)
         {
             //Shuffles the pack based on the type of shuffle
+            // 1. Fisher-Yates shuffle
+            // 2. Riffle shuffle
+            // 3. No shuffle
+
+            if (typeOfShuffle == 1)
+            {
+                System.Console.WriteLine("\nYou have chosen the Fisher-Yates Shuffle\n");
+                Random rand = new Random();
+                for (int i = 0; i < pack.Count; i++)
+                {
+                    int j = rand.Next(i, pack.Count);
+                    (pack[j], pack[i]) = (pack[i], pack[j]);
+                }
+                return true;
+            }
+            else if (typeOfShuffle == 2)
+            {
+                System.Console.WriteLine("\nYou have chosen the Riffle Shuffle\n");
+                // need to divide the pack of cards into two sets
+
+                int i = 0;
+                while (i < (pack.Count / 2))
+                {
+                    split_deck.Add(pack[i]);
+                    split_deck.Add(pack[i + (pack.Count / 2)]);
+                    i++;
+                }
+                pack = split_deck;
+                return true;
+            }
+
+            else if (typeOfShuffle == 3)
+            {
 
+                System.Console.WriteLine("\nYou have chosen not to shuffle the deck of cards\n");
+                return true;
+            }
+            else
+            {
+                return false;

```

Comment:
Good use of error handling to make sure that if the input is not one that is selected it will return false